packages/grid/x-data-grid/src/components/base/GridFooterPlaceholder.tsx

```diff
@@ -8,5 +8,7 @@ export function GridFooterPlaceholder() {
     return null;
   }
 
-  return <rootProps.slots.footer {...rootProps.slotProps?.footer} />;
+  return (
+    <rootProps.slots.footer {...(rootProps.slotProps?.footer as any) /* FIXME: typing error */} />
```

Comment:
This one is interesting.
The reason it's failing without `as any` is this override in the docs examples:
https://github.com/mui/mui-x/blob/58c5fe87e6034b650146c69258f41b4f30cdefaa/docs/data/data-grid/components/CustomFooter.tsx#L10-L14

I there something we can do to isolate overrides in `docs` from the packages?

Reply:
The way it's setup, the docs is importing the grid through relative imports (behind aliased paths). If the docs were importing code as regular packages, then the `skipLibCheck` flag would do that iiuc, but it doesn't apply here since it's just regular files. I think it could probably be done, but I'm not sure how long it will take to setup, so I'd skip it for this PR.